From linux-usb-devel-bounces@lists.sourceforge.net  Sat Sep  8 07:46:20 2007
From: Nathael Pajani <nathael.pajani@cpe.fr>
Date: Tue, 04 Sep 2007 11:46:23 +0200
Subject: USB: fix linked list insertion bugfix for usb core
To: Randy Dunlap <rdunlap@xenotime.net>
Cc: linux-usb-devel@lists.sourceforge.net
Message-ID: <46DD296F.4020501@cpe.fr>

From: Nathael Pajani <nathael.pajani@cpe.fr>

This patch fixes the order of list_add_tail() arguments in
usb_store_new_id() so the list can have more than one single element.

Signed-off-by: Nathael Pajani <nathael.pajani@cpe.fr>
Cc: stable <stable@kernel.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/core/driver.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/core/driver.c
+++ b/drivers/usb/core/driver.c
@@ -60,7 +60,7 @@ ssize_t usb_store_new_id(struct usb_dyni
 	dynid->id.match_flags = USB_DEVICE_ID_MATCH_DEVICE;
 
 	spin_lock(&dynids->lock);
-	list_add_tail(&dynids->list, &dynid->node);
+	list_add_tail(&dynid->node, &dynids->list);
 	spin_unlock(&dynids->lock);
 
 	if (get_driver(driver)) {
